### PR TITLE
BSP-2956 Add lesshint linting

### PR DESCRIPTION
**Pull request by rhino88**
This PR adds less linting configured by the `.lesshintrc`. Each enabled linter is explicitly marked as an "error" instead of a warning. For traceability, even default lesshint values are explicitly stated in the `.lesshintrc`.

Example output:

image

The [current gulp-lesshint feature](https://github.com/lesshint/gulp-lesshint/blob/master/index.js#L110) to `failOnError()` only provides information on how many errors were detected, and does not provide information on individual errors. A subsequent PR will look to create a custom reporter for this, use lesshint directly, or use gulp-plumber to provide better error output and build failure.

References: [18f styleguide](https://pages.18f.gov/frontend/ ), [Airbnb styleguide](https://github.com/airbnb/css)

### CODE REVIEW

File: js/task/lint.js [hunk 1]
```diff
@@ -24,8 +24,13 @@ module.exports = (styleguide, gulp) => {
     },
 
     less: () => {
-      return gulp.src(path.join(styleguide.path.root(), 'styleguide/All.less'))
-        .pipe(lesshint())
+      return gulp.src(path.join(styleguide.path.root(), 'styleguide/**/*.less'))
+        .pipe(lesshint({
+          configPath: path.join(
```
Comment (jpencola) [inline js/task/lint.js]: Minor nitpick: Please flatten this down to a single-line for consistency

**Review by jpencola — CHANGES_REQUESTED** [review]

**Review by jpencola — APPROVED** [review]

**Review by jpencola — COMMENTED** [review]
@rhino88 What do you think about bumping up the maxCharPerLine to `100`?

**Review by jpencola — COMMENTED** [review]
@rhino88 Using our container-query approach, I think it's going to be common to go beyond 3 levels deep. What do you think about bumping depthLevel up to `4`?

**Comment by trhoppe** [comment]
Won't the 3 levels deep be inside the mixing and not in the container query?

**Review by jpencola — COMMENTED** [review]
@trhoppe You're right, it seems that the linter doesn't add up the level(s) the mix-in is being used within.

**Comment by rhino88** [comment]
Increased maxCharPerLine to 100, per @jpencola

**Comment by jpencola** [comment]
@rhino88 / @trhoppe After running this against the Express codebase and discussing the potential for depth levels > 3 when using `extend`, I've increased the depthLevel to 5.

**Comment by jpencola** [comment]
@rhino88 I'm seeing the linter report a `depthLevel` error:
`There shouldn't be more than '5' levels deep from the style's parent, check the children's depth`
 given this selector:

```less
.Foo {
    &-one {
        color: red;
    }

    &-two {
        color: red;
    }

    &-three {
        color: red;
    }

    &-four {
        color: red;
    }

    &-five {
        color: red;
    }
}
```

Do you notice this too?

**Comment by rhino88** [comment]
Temporarily disabled the depth level check. Hoping we can open a github issue and reenable later.

**Pull request merged**